Remove test_groupby_agg_extension (marked xfail)

MichaelTiemannOSC opened the pull request:
Remove special `test_groupby_agg_extension` in `test_json.py` as we can now pass the test generally with these fixes to `GroupBy.first` and `GroupBy.last`.

- [x] closes #39098
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

MichaelTiemannOSC committed:
Remove test_groupby_agg_extension (marked xfail)

Remove special `test_groupby_agg_extension` in `test_json.py` as we can now pass the test generally with these fixes to `GroupBy.first` and `GroupBy.last`.

Signed-off-by: Michael Tiemann <72577720+MichaelTiemannOSC@users.noreply.github.com>

MichaelTiemannOSC committed:
Update v2.1.0.rst

Fix sort order.

Signed-off-by: Michael Tiemann <72577720+MichaelTiemannOSC@users.noreply.github.com>

MichaelTiemannOSC commented:
In the course of winnowing away unnecessary changes in a PR related to extension arrays, I found that this was one useful result of that work.  I'm re-submitting as a fresh PR that specifically fixes a single reported bug, following guidance in that bug report, and what I've learned along the way.

mroeschke commented on pandas/core/groupby/groupby.py:
```diff
@@ -3338,7 +3338,7 @@ def last(x: Series):
                 arr = x.array[notna(x.array)]
                 if not len(arr):
-                    return np.nan
+                    x.array.dtype.na_value
```
```suggestion
                    return x.array.dtype.na_value
```

MichaelTiemannOSC replied:
Whoops!  You are 100% correct there.

mroeschke commented on doc/source/whatsnew/v2.1.0.rst:
```diff
@@ -643,6 +643,8 @@ Other
 - Bug in :meth:`DataFrame.reindex` with a ``fill_value`` that should be inferred with a :class:`ExtensionDtype` incorrectly inferring ``object`` dtype (:issue:`52586`)
 - Bug in :meth:`DataFrame.shift` and :meth:`Series.shift` and :meth:`DataFrameGroupBy.shift` when passing both "freq" and "fill_value" silently ignoring "fill_value" instead of raising ``ValueError`` (:issue:`53832`)
 - Bug in :meth:`DataFrame.shift` with ``axis=1`` on a :class:`DataFrame` with a single :class:`ExtensionDtype` column giving incorrect results (:issue:`53832`)
+- Bug in :meth:`GroupBy.first` the helper function should return the NA value of the EA if all values are NA (:issue:`39098`)
+- Bug in :meth:`GroupBy.last` the helper function should return the NA value of the EA if all values are NA (:issue:`39098`)
```
```suggestion
- Bug in :meth:`GroupBy.first` and :meth:`GroupBy.last` where an empty group would return ``np.nan`` instead of a an ExtensionArray's NA value (:issue:`39098`)
```

MichaelTiemannOSC replied:
Will do.

MichaelTiemannOSC committed:
Apply suggestions from code review

Incorporate suggestions from @mroeschke (consolidating docs and fixing a malformed return statement).

Signed-off-by: Michael Tiemann <72577720+MichaelTiemannOSC@users.noreply.github.com>

Co-authored-by: Matthew Roeschke <10647082+mroeschke@users.noreply.github.com>

mroeschke commented:
Thanks @MichaelTiemannOSC
